lightning/src/routing/router.rs

```diff
@@ -623,19 +626,26 @@ fn compute_fees(amount_msat: u64, channel_fees: RoutingFees) -> Option<u64> {
 /// [`Event::PaymentPathFailed`]: crate::util::events::Event::PaymentPathFailed
 pub fn find_route<L: Deref, S: Score>(
 	our_node_pubkey: &PublicKey, route_params: &RouteParameters, network: &NetworkGraph,
-	first_hops: Option<&[&ChannelDetails]>, logger: L, scorer: &S
+	first_hops: Option<&[&ChannelDetails]>, logger: L, scorer: &S, random_seed_bytes: &[u8; 32]
 ) -> Result<Route, LightningError>
 where L::Target: Logger {
-	get_route(
-		our_node_pubkey, &route_params.payment_params, network, first_hops, route_params.final_value_msat,
-		route_params.final_cltv_expiry_delta, logger, scorer
-	)
+	let network_graph = network.read_only();
+	match get_route(
+		our_node_pubkey, &route_params.payment_params, &network_graph, first_hops, route_params.final_value_msat,
+		route_params.final_cltv_expiry_delta, logger, scorer, random_seed_bytes
+	) {
+		Ok(mut route) => {
+			add_random_cltv_offset(&mut route, &route_params.payment_params, &network_graph, random_seed_bytes);
+			Ok(route)
+		},
+		Err(err) => Err(err),
+	}
 }
 
 pub(crate) fn get_route<L: Deref, S: Score>(
-	our_node_pubkey: &PublicKey, payment_params: &PaymentParameters, network: &NetworkGraph,
-	first_hops: Option<&[&ChannelDetails]>, final_value_msat: u64, final_cltv_expiry_delta: u32, 
-	logger: L, scorer: &S
+	our_node_pubkey: &PublicKey, payment_params: &PaymentParameters, network_graph: &ReadOnlyNetworkGraph,
+	first_hops: Option<&[&ChannelDetails]>, final_value_msat: u64, final_cltv_expiry_delta: u32,
+	logger: L, scorer: &S, random_seed_bytes: &[u8; 32]
```

Comment:
`random_seed_bytes` is not needed here since it is only used in `add_random_cltv_offset`.

Reply:
That is correct, but the idea was to leave it in `get_route` for now and use it to implement the ['real random shuffle' TODO](https://github.com/lightningdevkit/rust-lightning/blob/main/lightning/src/routing/router.rs#L1439) in a follow-up PR. I now prefixed it with an underscore to disable the warning, but let me know if you'd rather keep it out of `get_route` altogether in this PR.

Reply:
hey @tnull is it okay if I take a stab at implementing the real random shuffle? I believe it may be related to this issue https://github.com/lightningdevkit/rust-lightning/issues/869

Reply:
> hey @tnull is it okay if I take a stab at implementing the real random shuffle? I believe it may be related to this issue #869

Ah, thank you for the interest! This issue indeed seems related. However, I’m already mostly done with a first draft and plan to open a PR in the next few days.